AlphaWallet/Settings/ViewModels/ExportJsonKeystorePasswordViewModel.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  ExportJsonKeystorePasswordViewModel.swift
+//  AlphaWallet
+//
+//  Created by Jerome Chan on 2/12/21.
+//
+
+import Foundation
+
+typealias ExportJsonKeystorePasswordViewModelCompletion = (Result<String, KeystoreError>) -> Void
+
+class ExportJsonKeystorePasswordViewModel {
+    private let validator: StringValidator
+    private let keystore: Keystore
+
+    init(keystore: Keystore) {
+        self.validator = StringValidator(rules: [
+            .lengthMoreThanOrEqualTo(6),
+            .canOnlyContain(CharacterSet.alphanumerics)
+        ])
+        self.keystore = keystore
+    }
+
+    init(keystore: Keystore, validator: StringValidator) {
+        self.validator = validator
+        self.keystore = keystore
+    }
+
+    func validate(password: String) -> StringValidatorResult {
+        return validator.validate(string: password)
+    }
+
+    func containsIllegalCharacters(password: String) -> Bool {
+        return validator.containsIllegalCharacters(string: password)
+    }
+
+    func computeHash(password: String, completion: @escaping ExportJsonKeystorePasswordViewModelCompletion) {
+        if keystore.isHdWallet(wallet: keystore.currentWallet) {
+            // TODO: Convert from HDWallet into Non-HDWallet then export
```

Comment:
Do remember to point out this in future PRs, too easy to miss during review

Reply:
I'll post an issue once this pull request is merged.

Reply:
Ok great. Thanks. But in such a case, would be good to hide the new functionality behind a feature flag in the same PR (described somewhere else in this PR). We release quite aggressively/randomly, so have to make sure `master` is somewhat ready anytime.